hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java

```diff
@@ -794,13 +795,30 @@ void shutdownLongCompactions(){
   }
 
   public void clearLongCompactionsQueue() {
+    removeFromFilesCompacting(longCompactions);
     longCompactions.getQueue().clear();
   }
 
   public void clearShortCompactionsQueue() {
+    removeFromFilesCompacting(shortCompactions);
     shortCompactions.getQueue().clear();
   }
 
+  private void removeFromFilesCompacting(ThreadPoolExecutor compactor) {
+    for (Runnable runnable : compactor.getQueue()) {
+      if (!(runnable instanceof CompactionRunner)) {
+        continue;
+      }
+      CompactionRunner runner = (CompactionRunner) runnable;
+      // for system compaction, files selection will be delayed until the compaction task
+      // actually runs, so compaction context is null for system compaction
+      if (runner.compaction != null) {
+        Collection<HStoreFile> files = runner.compaction.getRequest().getFiles();
```

Comment:
Before obtaining the file list, you should check whether the compacting has already selected the files

Reply:
thanks for your advise. I will fix it.

Reply:
Is this addressed?